meshtastic/interfaces/ble/lifecycle_disconnect_runtime.py

```
@@ -0,0 +1,551 @@
+                exc_info=True,
+            )
+            _close_inline()
+            raise
+        except Exception:  # noqa: BLE001 - cleanup must not abort disconnect flow
+            logger.warning(
+                "Failed to start async BLE client close; closing inline.",
+                exc_info=True,
+            )
+            _close_inline()
+
+    def _execute_disconnect_side_effects(
```

Ruff (0.15.6)

[warning] Too many branches (19 > 12)

(PLR0912)